rdflib/term.py

```diff
@@ -1800,8 +1811,6 @@ def _castPythonToLiteral(
 
 _check_well_formed_types: Dict[URIRef, Callable[[Union[str, bytes], Any], bool]] = {
     URIRef(_XSD_PFX + "boolean"): _well_formed_boolean,
-    URIRef(_XSD_PFX + "integer"): _well_formed_by_value,
-    URIRef(_XSD_PFX + "long"): _well_formed_by_value,
```

Comment:
Removed these as the default is now to do `_well_formed_by_value` unless there is a specific checker registered, mainly because a literal can only be well typed if a literal value can be associated with it AFAIU from this:

https://www.w3.org/TR/rdf11-concepts/#section-Graph-Literal

> If the literal's [datatype IRI](https://www.w3.org/TR/rdf11-concepts/#dfn-datatype-iri) is in the set of [recognized datatype IRIs](https://www.w3.org/TR/rdf11-concepts/#dfn-recognized-datatype-iris), let d be the [referent](https://www.w3.org/TR/rdf11-concepts/#dfn-referent) of the datatype IRI.
> - If the literal's [lexical form](https://www.w3.org/TR/rdf11-concepts/#dfn-lexical-form) is in the [lexical space](https://www.w3.org/TR/rdf11-concepts/#dfn-lexical-space) of d, then the literal value is the result of applying the [lexical-to-value mapping](https://www.w3.org/TR/rdf11-concepts/#dfn-lexical-to-value-mapping) of d to the [lexical form](https://www.w3.org/TR/rdf11-concepts/#dfn-lexical-form).
> - Otherwise, the literal is ill-typed and no literal value can be associated with the literal. Such a case produces a semantic inconsistency but is not syntactically ill-formed. Implementations must accept ill-typed literals and produce RDF graphs from them. Implementations may produce warnings when encountering ill-typed literals.